api/src/main/java/uk/gov/hmcts/payment/api/controllers/ServiceRequestController.java

```diff
@@ -139,7 +139,17 @@
         String responseJson;
         try {
             serviceRequestPaymentBo = serviceRequestDomainService.addPayments(serviceRequest, serviceRequestPaymentDto);
-            HttpStatus httpStatus = serviceRequestPaymentBo.getStatus().equalsIgnoreCase(FAILED) ? HttpStatus.PAYMENT_REQUIRED : HttpStatus.CREATED; //402 for failed Payment scenarios
+            HttpStatus httpStatus;
+            if(serviceRequestPaymentBo.getError() != null && serviceRequestPaymentBo.getError().getErrorCode().equals("CA-E0004")) {
```

Comment:
Can you reverse this to 
("CA-E0004").equals (serviceRequestPaymentBo.getError().getErrorCode()) to avoid NullPointerException on getErrorCode